index.ts

```diff
@@ -0,0 +1,543 @@
+import type { Inflection, Options, SchemaBuilder } from "graphile-build";
+import type {
+  PgClass,
+  PgConstraint,
+  PgEntity,
+  PgProc,
+} from "graphile-build-pg";
+
+function fixCapitalisedPlural(fn: (this: Inflection, str: string) => string) {
+  return function (this: Inflection, str: string) {
+    const original = fn.call(this, str);
+    return original.replace(/[0-9]S(?=[A-Z]|$)/g, (match) =>
+      match.toLowerCase()
+    );
+  };
+}
+
+function fixChangePlural(fn: (this: Inflection, str: string) => string) {
+  return function (this: Inflection, str: string) {
+    const matches = str.match(/([A-Z]|_[a-z0-9])[a-z0-9]*_*$/);
+    const index = matches ? matches.index! + matches[1].length - 1 : 0;
+    const suffixMatches = str.match(/_*$/);
+    const suffixIndex = suffixMatches!.index!;
```

Comment:
Assertions: every string matches this regexp, so suffixMatches is guaranteed; and index will exist because we're not using `/g`